compiler/transforms/dispatch_elementwise_mult.py

```diff
@@ -1,26 +1,29 @@
-from xdsl.dialects import builtin, func
-from dialects import linalg_extension
+from xdsl.dialects import builtin, linalg, arith
 from xdsl.ir import MLContext
 from xdsl.passes import ModulePass
 from xdsl.dialects.memref import MemRefType
 from xdsl.pattern_rewriter import (
-    GreedyRewritePatternApplier,
     PatternRewriteWalker,
     PatternRewriter,
     RewritePattern,
     op_type_rewrite_pattern,
 )
-from xdsl.traits import SymbolTable
 
 
 class AddLibraryCall(RewritePattern):
     @op_type_rewrite_pattern
-    def match_and_rewrite(self, op: linalg_extension.Mul, rewriter: PatternRewriter):
+    def match_and_rewrite(self, op: linalg.Generic, rewriter: PatternRewriter):
```

Comment:
Would be good to have some part of your reasoning in your PR here directly?
```suggestion
    def match_and_rewrite(self, op: linalg.Generic, rewriter: PatternRewriter):
        """Add a library call to a linalg generic that implements an elementwise multiplication (similar to linalg.mul).    Adding such an attribute to a linalg.mul op directly is not possible."""
```